src/main/scala/mesosphere/marathon/core/task/tracker/InstanceTracker.scala

```diff
@@ -32,6 +38,19 @@ trait InstanceTracker {
 
   def hasSpecInstancesSync(appId: PathId): Boolean
```

Comment:
Not necessarily in this diff but those `*Sync` methods are used almost exclusively in tests. Since you're cleaning up, wdyt about removing those completely?

Reply:
And remove the todo. Sure but not in this diff. I'll follow up.